### PR TITLE
Add UXV Component Assemblyline functionality

### PR DESCRIPTION
**Pull request by GDCloudstrike**
Adds UXV compat for the CoAl.  
WARNING: THIS WILL BREAK LOWER TIER PATTERNS THAT HAVE MORE THAN 64 OF THE SAME MATERIAL PART (these will be converted to fluid)

UXV parts are a bit trickier to handle than other tiers because of the fact that mhdcsm parts cannot just simply be converted to a fluid (because of their special EIC recipes).

That is why the CoAl recipe contains the appropriate amount of molten eternity and UHV circuit wraps to simulate the EIC recipe.
Furthermore this PR changes the stacks shown in NEI to uncapped ones because of space issues.

For some reason certain oredicts cannot exceed 64 items per stack (e.g. nanites or plates) so the neutronium nanites used in UXV part recipes are converted to gold nanites at a 16:1 ratio.

UXV Pump recipe as an example:
image

Casing recipe:
image

### CODE REVIEW

File: src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java [hunk 1]
```diff
@@ -363,4 +399,44 @@ private static List<ItemStack> getWrappedCircuits(ItemStack item, int total, Str
...
+                                GT_OreDictUnificator.get(OrePrefixes.circuit, Materials.Infinite, 1),
+                                (int) (total * multiplier),
+                                "circuitInfinite"));
+            }
+            stacks.addAll(multiplyAndSplitIntoStacks(item, total));
+        }
+        return stacks;
+    }
+
+    private static void addEternity(ArrayList<FluidStack> fluidInputs) {
```
Comment (BlueWeabo) [inline src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java:420]: If this does both Eternity and mhdcsm it should say. Maybe change the method name?

File: src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java [hunk 2]
```diff
@@ -363,4 +399,44 @@ private static List<ItemStack> getWrappedCircuits(ItemStack item, int total, Str
 
         return stacks;
     }
+
+    private static List<ItemStack> getMagnetoConversion(ItemStack item, int total) {
+        ArrayList<ItemStack> stacks = new ArrayList<>();
+        ItemData data = GT_OreDictUnificator.getAssociation(item);
+        if (data != null) {
```
Comment (BlueWeabo) [inline src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java:406]: any reason for not inverting the if here to remove an indent?

File: src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java [hunk 3]
```diff
@@ -139,11 +159,13 @@ private static void generateAsslineRecipes() {
                         }
                     }
                     fixedInputs = compactItems(fixedInputs, info.getRight());
-                    replaceIntoFluids(fixedInputs, fixedFluids, 128);
+                    replaceIntoFluids(fixedInputs, fixedFluids, 64);
```
Comment (BlueWeabo) [inline src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java:162]: can you make the 64 into a constant? I can't understand what it means from looking at it here. and if you want can do the other ones too.

File: src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java [hunk 4]
```diff
@@ -363,4 +399,44 @@ private static List<ItemStack> getWrappedCircuits(ItemStack item, int total, Str
...
+    }
+
+    private static void addEternity(ArrayList<FluidStack> fluidInputs) {
+        boolean eternity = false;
+        boolean mhdcsm = false;
+        int mhdcsmAmount = 0;
+
+        for (FluidStack fluidstack : fluidInputs) {
+            if (fluidstack.getFluid()
+                    .equals(FluidRegistry.getFluid("molten.magnetohydrodynamicallyconstrainedstarmatter"))) {
```
Comment (BlueWeabo) [inline src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java:427]: you can shorten this by putting the mhdcsm's fluid name into a constant. Maybe not with its full name and add a doc to the variable?

File: src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java (hunk 1 above)
Comment (GDCloudstrike) [reply src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java:420]: this adds eternity based on the amount of mhdcsm present

Comment (BlueWeabo) [reply src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java:420]: AHHH. haven't played in a while and I forgot.

**Review by BlueWeabo — COMMENTED** [review]
Approved from me on the code side. Looks good to me

**Review by LewisSaber — APPROVED** [review]

**Pull request merged**